pkgs/kde/gear/akonadi/default.nix

```diff
@@ -16,24 +26,35 @@ mkKdeDerivation {
     ./ignore-mysql-config-timestamp.patch
   ];
 
-  extraCmakeFlags = [
-    "-DMYSQLD_SCRIPTS_PATH=${lib.getBin mariadb}/bin"
-  ];
+  extraCmakeFlags =
+    [ "-DDATABASE_BACKEND=${lib.toUpper backend}" ]
+    ++ lib.optionals (backend == "mysql") [
+      "-DMYSQLD_SCRIPTS_PATH=${lib.getBin mariadb}/bin"
+    ]
+    ++ lib.optionals (backend == "postgres") [
+      "-DPOSTGRES_PATH=${lib.getBin postgresql}/bin"
+    ];
 
   extraNativeBuildInputs = [
     qttools
     shared-mime-info
   ];
-  extraBuildInputs = [
-    kaccounts-integration
-    accounts-qt
-    xz
-    mariadb
-  ];
+
+  extraBuildInputs =
+    [
+      kaccounts-integration
+      accounts-qt
+      xz
+    ]
+    ++ lib.optionals (backend == "mysql") [ mariadb ]
+    ++ lib.optionals (backend == "postgres") [ postgresql ]
+    ++ lib.optionals (backend == "sqlite") [ sqlite ];
 
   # Hardcoded as a QString, which is UTF-16 so Nix can't pick it up automatically
   postFixup = ''
     mkdir -p $out/nix-support
-    echo "${mariadb}" > $out/nix-support/depends
+     ${lib.optionalString (backend == "mysql") "echo '${mariadb}' >> $out/nix-support/depends"}
+     ${lib.optionalString (backend == "postgres") "echo '${postgresql}' >> $out/nix-support/depends"}
+     ${lib.optionalString (backend == "sqlite") "echo '${sqlite}' >> $out/nix-support/depends"}
```

Comment:
Probably don't need to explicitly add sqlite, since it's not a binary? Not sure about postgres, either.

Reply:
i am honestly not sure what its for, so i just did the same as for mysql.

Reply:
Postgres does need a reference: https://invent.kde.org/pim/akonadi/-/blob/master/src/server/storage/dbconfigpostgresql.cpp?ref_type=heads#L145

Reply:
i have re-added a reference for postgresql.